### PR TITLE
GEOMESA-2688 Fix heatmap rendering outside [-180,180] longitude

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Geometries are translated to overlap the rendering envelope
* Filters extracted from the envelope are split for AntiMeridian handling
* Translation logic consolidated in org.locationtech.geomesa.utils.geotools.RenderingGrid

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/iterators/DensityScan.scala [hunk 1]
```diff
@@ -203,235 +198,110 @@ object DensityScan extends LazyLogging {
...
 
   /**
     * Evaluates an arbitrary expression against the simple feature to return a weight
     */
-  private def getWeightFromExpression(e: Expression)(sf: SimpleFeature): Double = {
-    val d = e.evaluate(sf, classOf[java.lang.Double])
-    if (d == null) 0.0 else d
+  class WeightByExpression(e: Expression) extends Weigher {
+    override def weight(sf: SimpleFeature): Double = {
+      val d = e.evaluate(sf, classOf[java.lang.Double])
```
Comment (jnh5y) [inline geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/iterators/DensityScan.scala]: Does this have any issues if the type of the attribute is an int / long / not-a-double?

File: geomesa-utils/src/test/scala/org/locationtech/geomesa/utils/geotools/GeometryUtilsTest.scala [hunk 2]
```diff
@@ -37,5 +38,36 @@ class GeometryUtilsTest extends Specification {
         min must beLessThanOrEqualTo(max)
       }
     }
+
+    "split bounding box envelopes" >> {
+      val unchanged = Seq(
+        new ReferencedEnvelope(-180, 180, -90, 90, CRS_EPSG_4326),
```
Comment (jnh5y) [inline geomesa-utils/src/test/scala/org/locationtech/geomesa/utils/geotools/GeometryUtilsTest.scala]: What happens if a bbox covers and contains this one?  E.g., if the bbox is larger than the world?

File: geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/iterators/DensityScan.scala (hunk 1 above)
Comment (elahrvivaz) [reply geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/iterators/DensityScan.scala]: when you pass in classOf double, it tries to convert the result to a double, using the geotools Converters class.

File: geomesa-utils/src/test/scala/org/locationtech/geomesa/utils/geotools/GeometryUtilsTest.scala (hunk 2 above)
Comment (elahrvivaz) [reply geomesa-utils/src/test/scala/org/locationtech/geomesa/utils/geotools/GeometryUtilsTest.scala]: thanks, added a check and test for that.

**Pull request merged**